### PR TITLE
[tools/depends][target] Add brotli lib for curl

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Noticed that for our windows curl lib, we build and enable brotli - https://github.com/xbmc/kodi-deps/blob/17b83b0df1d5d3b33e4e0d106f125a454e2e363e/CMakeLists.txt#L452
This just makes this available for tools/depends targets

## Motivation and context
Parity with windows. We can go either way i guess, remove brotli use from windows, or enable for other platforms. I dont think theres any harm in just enabling for others. The build is small, and cmake based.

I have this based off #25071 with the curl bump as it was just easiest. Will remove 7275d66ff3f1fe039d6fdde0f4d067532fdff9f8 before merge

## How has this been tested?
Locally build aarch64 macos

Before
```
  curl version:     8.7.1
  SSL:              enabled (OpenSSL v3+)
  SSH:              no      (--with-{libssh,libssh2})
  zlib:             enabled
  brotli:           no      (--with-brotli)
  zstd:             no      (--with-zstd)
  GSS-API:          no      (--with-gssapi)
  GSASL:            no      (libgsasl not found)
```

After

```
  curl version:     8.7.1
  SSL:              enabled (OpenSSL v3+)
  SSH:              no      (--with-{libssh,libssh2})
  zlib:             enabled
  brotli:           enabled (libbrotlidec)
  zstd:             no      (--with-zstd)
  GSS-API:          no      (--with-gssapi)
  GSASL:            no      (libgsasl not found)
```

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by fuzzard** [comment]
Jenkins build this please

File: tools/depends/target/brotli/BROTLI-VERSION [hunk 1]
```diff
@@ -0,0 +1,8 @@
+LIBNAME=brotli
+VERSION=1.1.0
+#ARCHIVE=$(LIBNAME)-$(VERSION).tar.gz
```
Comment (wsnipex) [inline tools/depends/target/brotli/BROTLI-VERSION:3]: Todo

File: tools/depends/target/brotli/BROTLI-VERSION [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+ARCHIVE=v$(VERSION).tar.gz
+SHA512=6eb280d10d8e1b43d22d00fa535435923c22ce8448709419d676ff47d4a644102ea04f488fc65a179c6c09fee12380992e9335bad8dfebd5d1f20908d10849d9
+BYPRODUCT=libbrotlidec.a
+
+BASE_URL=https://github.com/google/brotli/archive/refs/tags
```
Comment (wsnipex) [inline tools/depends/target/brotli/BROTLI-VERSION:8]: Todo

**Review by wsnipex — COMMENTED** [review]
Looks good

File: cmake/modules/FindCurl.cmake [hunk 3]
```diff
@@ -38,6 +38,13 @@ if(${CURL_LIBRARY} MATCHES ".+\.a$" AND PC_CURL_STATIC_LDFLAGS)
   pkg_check_modules(PC_NGHTTP2 libnghttp2 QUIET)
   find_library(NGHTTP2_LIBRARY NAMES libnghttp2 nghttp2
                                HINTS ${PC_NGHTTP2_LIBDIR})
+
+  find_library(BROTLIDEC_LIBRARY NAMES brotlidec
+                                 HINTS ${PC_BROTLI_LIBDIR})
+  find_library(BROTLICOMMON_LIBRARY NAMES brotlicommon
+                                    HINTS ${PC_BROTLI_LIBDIR})
+  set(BROTLI_LIBRARIES ${BROTLIDEC_LIBRARY} ${BROTLICOMMON_LIBRARY})
```
Comment (fuzzard) [inline cmake/modules/FindCurl.cmake:46]: Id forgotten how a lot of our old stuff "discovered" libraries. This is the shitty way, but i have other work that redoes the curl find module completely, so as long as @wsnipex is ok with this in the interim.

The "new" way i'll be proposing will be https://github.com/fuzzard/xbmc/commit/2ce516d254401521145456fa44f66d36509745dc
It essentially moves to cmake for internal stuff (windows/android/apple), but has a fallback for using pkgconfig data for the additional libraries in the event of a static curl lib being found, therefore we dont need this manual explicit find library calls for nghttp2 and now brotli, we just rely on what the pkgconfig provides for additional link requirements

**Pull request merged**